pkg/api/mlflow/dao/repositories/run.go

```diff
@@ -156,18 +156,31 @@ func (r RunRepository) ArchiveBatch(ctx context.Context, ids []string) error {
 
 // Delete removes the existing models.Run from the db.
 func (r RunRepository) Delete(ctx context.Context, run *models.Run) error {
-	if err := r.db.WithContext(ctx).Model(&run).Delete(run).Error; err != nil {
-		return eris.Wrapf(err, "error deleting run with id: %s", run.ID)
-	}
-
-	return nil
+	return r.DeleteBatch(ctx, []string{run.ID})
 }
 
 // DeleteBatch removes existing models.Run from the db.
 func (r RunRepository) DeleteBatch(ctx context.Context, ids []string) error {
-	run := models.Run{}
-	if err := r.db.WithContext(ctx).Model(&run).Where("run_uuid IN ?", ids).Delete(run).Error; err != nil {
-		return eris.Wrapf(err, "error deleting existing runs with ids: %s", ids)
+	if err := r.db.Transaction(func(tx *gorm.DB) error {
+		// delete the rows
+		runs := []models.Run{}
+		tx.Clauses(clause.Returning{Columns: []clause.Column{{Name: "row_num"}}}).Where("run_uuid IN ?", ids).Delete(&runs)
+		if tx.Error != nil {
+			return eris.Wrapf(tx.Error, "error deleting existing runs with ids: %s", ids)
+		}
+
+		// verify deletion
+		if len(runs) != len(ids) {
+			return eris.Errorf("non-existent run id provided")
+		}
+
+		// renumber the remainder
+		if err := r.renumberRows(tx, getMinRowNum(runs)); err != nil {
+			return eris.Wrapf(err, "error renumbering runs.row_num")
+		}
+		return nil
+	}); err != nil {
+		return eris.Wrapf(err, "error renumbering runs.row_num")
```

Comment:
I think this error is about deleting runs, not renumbering them per se, as it's at the transaction level

Reply:
oops, fixed